scripts/link_linter.js

```diff
@@ -0,0 +1,191 @@
+const parse = require("node-html-parser");
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * Recursively search for html files and add them to the result list.
+ *
+ * @param {string[]} files A list of files to search through.
+ * @param {string[]} result A list of complete file paths to html files.
+ */
+function fetchFilePaths(files, result = []) {
+    // Grab the first file in the list and build its full path.
+    const file = files[0];
```

Comment:
What if `files` is undefined/empty? Rather than just throwing with an index out of bounds error here, perhaps we should just return `if (!files || files.length === 0)`?

Reply:
Added a check at the top of the function for this.